ADApp/pluginSrc/NDFileHDF5Dataset.cpp

```diff
@@ -188,8 +254,43 @@ asynStatus NDFileHDF5Dataset::writeFile(NDArray *pArray, hid_t datatype, hid_t d
               fileName, functionName);
     return asynError;
   }
+
   // Write the data to the hyperslab.
-  hdfstatus = H5Dwrite(this->dataset_, datatype, dataspace, fspace, H5P_DEFAULT, pArray->pData);
+  if (H5_VERSION_GE(1, 8, 11) && verifyChunking(pArray) == asynSuccess) {
+    // The chunking and compression settings match - use direct chunk write
+    asynPrint(this->pAsynUser_, ASYN_TRACE_FLOW,
+              "%s::%s NDArray correctly chunked. Using direct chunk write\n",
+              fileName, functionName);
+    #if H5_VERSION_GE(1, 8, 11)
```

Comment:
You test for H5 version greater than 1.8.11 here and then inside the if statement you test again with a #define.  That code can never be executed.